roxie/ccd/ccdserver.cpp

```diff
@@ -9270,6 +9279,18 @@ class CRoxieServerThroughSpillActivity : public CRoxieServerActivity
         CRoxieServerActivity::stop();
     };
 
+    virtual bool isSimpleSink() const override
+    {
+        // Note - there's a possibility that two threads arriving here very close together will result in parent isSimpleSink being called twice,
+        // but that is harmless
+        if (!simpleSinkCalculated)
+        {
+            simpleSinkCalculated = true;
```

Comment:
These lines need to be swapped - otherwise another thread will read an invalid value.